### ISSUE TITLE
Keras consistency issue

### ISSUE THREAD

**Issue by KiranBaktha** [fchollet/keras#6161]
Hi guys,
I am using Keras 2.0.2 with TensorFlow backend. I have read the previous issues in Keras giving different results at every run. So I set my Numpy seed initially like this : 

        np.random.seed(1337)
        import tensorflow as tf
        tf.set_random_seed(1337)

I want to run an LSTM layer so I also initialized all the initializers to a fixed seed like this : 

     lstm = LSTM(300,activation='tanh', recurrent_activation='sigmoid', recurrent_initializer=keras.initializers.RandomNormal(mean=0.0, stddev=0.05, seed=33), bias_initializer=keras.initializers.RandomNormal(mean=0.0, stddev=0.05, seed=33), kernel_initializer=keras.initializers.RandomNormal(mean=0.0, stddev=0.05, seed=33))(Embeddings)

I have also set " shuffle = False "    while fitting the model so that the input does not get shuffled and set the number of epochs to be 1. When I check the LSTM weights, I get two different weights for two different runs. Could you please let me know how to solve this issue?

Thanks in advance.
@fchollet 
@mbhenry 
@dragoon 
@lukeyeager

**Comment by fchollet** [comment]
Is this on GPU?

**Comment by KiranBaktha** [comment]
@fchollet  No I am running it on CPU. The CPU I am using is Intel i7 7th Gen.

**Comment by KiranBaktha** [comment]
Hi guys, so I managed to get the initial weights of the layers to be the same in every run. But after compiling the model the weights are different which gives different accuracy. Is there any reason behind this randomness? I have even not considered dropouts. 
@fchollet @mbhenry @dragoon @lukeyeager
